include/ensmallen_bits/cmaes/pop_cmaes.hpp

```diff
@@ -0,0 +1,175 @@
+ *   author={Hansen, Nikolaus},
+ *   booktitle={Proceedings of the 11th annual conference companion on genetic and evolutionary computation conference: late breaking papers},
+ *   pages={2389--2396},
+ *   year={2009}}
+ * @endcode
+ *
+ * BI-Population CMA-ES can efficiently handle separable, multimodal, and weak
+ * structure functions across various dimensions, as demonstrated in the 
+ * comprehensive results of the BBOB-2009 function testbed. The optimizer
+ * utilizes an interlaced multistart strategy to balance between broad 
+ * exploration and intensive exploitation, adjusting population sizes and 
+ * step-sizes dynamically.
+ */
+template<typename SelectionPolicyType = FullSelection,
+         typename TransformationPolicyType = EmptyTransformation<>,
+         bool UseBIPOPFlag = true>
+class POP_CMAES : public CMAES<SelectionPolicyType, TransformationPolicyType>
+{
+ public:
+  /**
+   * Construct the POP-CMA-ES optimizer with the given parameters.
+   * Other than the same CMA-ES parameters, it also adds the maximum number of 
+   * restarts, the increase in population factor, the maximum number of 
+   * evaluations, as well as a flag indicating to use BIPOP or not.
+   * The suggested values are not necessarily good for the given problem, so it
+   * is suggested that the values used be tailored to the task at hand. The
+   * maximum number of iterations refers to the maximum number of points that
+   * are processed (i.e., one iteration equals one point; one iteration does not
+   * equal one pass over the dataset).
+   * 
+   * @param lambda The initial population size (0 use the default size).
+   * @param transformationPolicy Instantiated transformation policy used to
+   *    map the coordinates to the desired domain.
+   * @param batchSize Batch size to use for the objective calculation.
+   * @param maxIterations Maximum number of iterations allowed.
+   * @param tolerance Maximum absolute tolerance to terminate algorithm.
+   * @param selectionPolicy Instantiated selection policy used to calculate the
+   *     objective.
+   * @param stepSize Starting sigma/step size (will be modified).
+   * @param populationFactor The factor by which population increases
+   *     after each restart.
+   * @param maxRestarts Maximum number of restarts.
+   * @param maxFunctionEvaluations Maximum number of function evaluations.
+   */
+  POP_CMAES(const size_t lambda = 0,
+           const TransformationPolicyType& transformationPolicy = 
+              TransformationPolicyType(),
+           const size_t batchSize = 32,
+           const size_t maxIterations = 1000,
+           const double tolerance = 1e-5,
+           const SelectionPolicyType& selectionPolicy = SelectionPolicyType(),
+           double stepSize = 0,
+           const double populationFactor = 2,
+           const size_t maxRestarts = 9,
+           const size_t maxFunctionEvaluations = 1e9);
```

Comment:
```suggestion
            const TransformationPolicyType& transformationPolicy = 
                 TransformationPolicyType(),
            const size_t batchSize = 32,
            const size_t maxIterations = 1000,
            const double tolerance = 1e-5,
            const SelectionPolicyType& selectionPolicy = SelectionPolicyType(),
            double stepSize = 0,
            const double populationFactor = 2,
            const size_t maxRestarts = 9,
            const size_t maxFunctionEvaluations = 1e9);
```
Super trivial style comment that I saw when looking through the new design.  It looks great to me :+1: